python-checks/src/test/resources/checks/torchModuleModeShouldBeSetAfterLoadingCheck.py

```diff
@@ -7,26 +7,23 @@ def noncompliant():
    #^^^^^^^^^^^^^^^^^^^^^
     ...
 
-def noncompliant(model):
-    model.load_state_dict(torch.load('model_weights.pth')) # Noncompliant
-
-def noncompliant():
-    model.load_state_dict(torch.load('model_weights.pth')) # Noncompliant
+class CustomModule(torch.nn.Module):
+    pass
 
 def noncompliant():
-    get_model().load_state_dict(torch.load('model_weights.pth')) # Noncompliant
-
-def noncompliant(model):
-    weights = torch.load('model_weights.pth')
-    weights2 = weights
-    model.load_state_dict(weights2) # Noncompliant
+    model = CustomModule(...)
+    model.load_state_dict(torch.load('model_weights.pth')) # false negative
+    ...
```

Comment:
I did not notice first, but we are missing a few tests.
```
load = torch.load(...)
model.load_state_dict(load)
```
as well as a check on the recursion. I think we could limit the recursion to 3.

Reply:
I removed the recursion entirely. Checking if the parameter is torch.load doesn't make sense anymore since the type of the model can be properly resolved